lib/terraforming/resource/route53_record.rb

```diff
@@ -66,18 +73,32 @@ def record_sets_of(hosted_zone)
       end
 
       def records
-        hosted_zones.map do |hosted_zone|
+        to_return = hosted_zones.map do |hosted_zone|
           record_sets_of(hosted_zone).map { |record| { record: record, zone_id: zone_id_of(hosted_zone) } }
         end.flatten
+        count = {}
+        dups = to_return.group_by{ |record| module_name_of(record[:record], nil) }.select { |k, v| v.size > 1 }.map(&:first)
+        to_return.each do |r|
+          module_name = module_name_of(r[:record], nil)
+          if dups.include?(module_name)
+            if count[module_name]
+              count[module_name] = count[module_name] + 1
+            else
+              count[module_name] = 0
+            end
+            r[:counter] = count[module_name]
+          end
+        end
+        to_return
       end
 
       # TODO(dtan4): change method name...
       def name_of(dns_name)
         dns_name.gsub(/\.\z/, "")
       end
 
-      def module_name_of(record)
-        normalize_module_name(name_of(record.name) + "-" + record.type)
+      def module_name_of(record, counter)
```

Comment:
It appears as though each resource type has its own `module_name_of()` method, so changing how the `route53_record` one works should be independent of the other ones.